code/game/mecha/equipment/mecha_equipment.dm

```diff
@@ -82,8 +78,8 @@
 /obj/item/mecha_parts/mecha_equipment/proc/start_cooldown()
 	set_ready_state(0)
 	chassis.use_power(energy_drain)
-	sleep(equip_cooldown)
-	set_ready_state(1)
+	spawn(equip_cooldown)
+		set_ready_state(1)

```

Comment:
This fixes an issue with mech ranged attack armor equipment not reacting correctly when hit by a fast burst of bullets.